From 47c73024739a65614c6a0ddcdd1c08239798044a Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Thu, 4 Oct 2012 17:36:57 -0400
Subject: [PATCH] slave: only set up autologin conversations

Right now any time any session conversation is started,
we run its setup call if autologin is enabled. We should only do this
for autologin sessions, otherwise, we may repeat work done by
the greeter.

https://bugzilla.gnome.org/show_bug.cgi?id=682467
---
 daemon/gdm-simple-slave.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/daemon/gdm-simple-slave.c b/daemon/gdm-simple-slave.c
index 9bc8522..4bbf503 100644
--- a/daemon/gdm-simple-slave.c
+++ b/daemon/gdm-simple-slave.c
@@ -485,8 +485,12 @@ on_session_conversation_started (GdmSession       *session,
         char    *username;
         int      delay;
 
+        g_debug ("GdmSimpleSlave: session conversation started for service %s", service_name);
+
+        if (g_strcmp0 (service_name, "gdm-autologin") != 0) {
+                return;
+        }
 
-        g_debug ("GdmSimpleSlave: session conversation started");
         enabled = FALSE;
         gdm_slave_get_timed_login_details (GDM_SLAVE (slave), &enabled, &username, &delay);
         if (! enabled) {
-- 
1.8.0.2

